app/routes/exercises.rb

```diff
@@ -66,6 +66,28 @@ class Exercises < Core
         redirect '/dashboard'
       end
 
+      post '/exercises/:key/request_help' do |key|
+        exercise = UserExercise.find_by_key(key)
+        unless current_user.owns?(exercise)
+          flash[:notice] = "Only the author may request help."
+          redirect '/'
+        end
+        exercise.request_help!(current_user)
+        flash[:success] = "You have requested for help on this submission"
+        redirect "/submissions/%s" % exercise.submissions.last.key
+      end
+
+      post '/exercises/:key/remove_help' do |key|
+        exercise = UserExercise.find_by_key(key)
+        unless current_user.owns?(exercise)
+          flash[:notice] = "Only the author may remove help."
+          redirect '/'
+        end
+        exercise.remove_help!(current_user)
+        flash[:success] = "Requested help has been removed"
```

Comment:
"Help request has been removed." or "Help request has been cancelled." or "Your request for help has now been removed."